components/de_amp/browser/de_amp_url_loader.cc

```diff
@@ -0,0 +1,399 @@
+/* Copyright 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/de_amp/browser/de_amp_url_loader.h"
+
+#include <memory>
+#include <string>
+#include <utility>
+
+#include "base/bind.h"
+#include "base/metrics/histogram_macros.h"
+#include "base/no_destructor.h"
+#include "base/task/post_task.h"
+#include "base/task/thread_pool.h"
+#include "brave/components/de_amp/browser/de_amp_service.h"
+#include "brave/components/de_amp/browser/de_amp_throttle.h"
```

Comment:
Both the throttle and the loader include a header to each other.
We use only `Resume()` from the throttle, better remove this include and use a callback instead.

Reply:
Makes sense, do you have an example I could take a look at?

Reply:
I mean simply passing `base::RepeatingClosure resume_callback`  to ctor  (instead for `DeAmpThrottle*`). This allow sto remove this include and makes the code/tests more clear.

Reply:
This would (now, after the refactoring) also change speedreader code. Is that okay?

Reply:
we agreed not to make any structural changes to speedreader so I think if this is inherited from that code we do a follow-up to fix